### PR TITLE
[TF] Implement _TensorFlowDataTypeCompatible protocol

### PR DESCRIPTION
**Pull request by njjiang**
This patch implements a new protocol `_TensorFlowDataTypeCompatible` which `AccelerableByTensorFlow` refines.

### CODE REVIEW

File: stdlib/public/TensorFlow/TensorHandle.swift [hunk 1]
```diff
@@ -108,7 +108,7 @@ extension TensorHandle : TensorSendableReceivable {
     TF_DeleteTensor(cTensor!)
     if _RuntimeConfig.printsDebugLog {
       debugLog("The received tensor of id \(tensorId) has content:")
-      dumpTensorContent(tensorHandle.cTensorHandle, Scalar.self)
+      // dumpTensorContent(tensorHandle.cTensorHandle, Scalar.self)
```
Comment (rxwei) [inline stdlib/public/TensorFlow/TensorHandle.swift]: To properly support `dumpTensorContents`, you should make `TensorHandle` **conditionally** conform to `TensorSendableReceivable` where `Scalar` conforms to `AccelerableByTensorFlow` on line 93. After doing that, you should be able to uncomment these lines.

File: stdlib/public/TensorFlow/TensorHandle.swift [hunk 2]
```diff
@@ -90,7 +90,8 @@ internal extension TensorHandle {
   }
 }
 
-extension TensorHandle : TensorSendableReceivable {
+extension TensorHandle : TensorSendableReceivable where
```
Comment (rxwei) [inline stdlib/public/TensorFlow/TensorHandle.swift]: Nit: put the `where` on the next line.

**Comment by njjiang** [comment]
Thanks for the review! I have implemented those changes.

File: stdlib/public/TensorFlow/Utilities.swift [hunk 3]
```diff
@@ -290,7 +290,7 @@ public func _hostOp<Scalar>(_ x: Tensor<Scalar>) {
 }
 
 @inline(never)
-public func _hostOp<Scalar>(_ x: TensorHandle<Scalar>) {
+public func _hostOp<Scalar : AccelerableByTensorFlow>(_ x: TensorHandle<Scalar>) {
```
Comment (rxwei) [inline stdlib/public/TensorFlow/Utilities.swift]: Fit this line within 80 columns. In this case, you can put a `where Scalar : AccelerableByTensorFlow` clause at the end.

**Review by rxwei — APPROVED** [review]
Thanks for tackling this!

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow seriously

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@njjiang Looks like CI is completely dead on this PR. Would you mind starting a new one?

**Pull request closed without merging**